arm/Microsoft.Network/virtualNetworks/subnets/readme.md

```diff
@@ -0,0 +1,73 @@
+| `Microsoft.Network/virtualNetworks/subnets` | 2021-03-01 |
+
+## Parameters
+
+| Parameter Name | Type | Default Value | Possible Values | Description |
+| :-- | :-- | :-- | :-- | :-- |
+| `addressPrefix` | string |  |  | Required. The address prefix for the subnet. |
+| `addressPrefixes` | array | `[]` |  | Optional. List of address prefixes for the subnet. |
+| `applicationGatewayIpConfigurations` | array | `[]` |  | Optional. Application gateway IP configurations of virtual network resource. |
+| `cuaId` | string |  |  | Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered |
+| `delegations` | array | `[]` |  | Optional. The delegations to enable on the subnet |
+| `ipAllocations` | array | `[]` |  | Optional. Array of IpAllocation which reference this subnet |
+| `name` | string |  |  | Optional. The Name of the subnet resource. |
+| `natGatewayName` | string |  |  | Optional. The name of the NAT Gateway to use for the subnet |
+| `networkSecurityGroupName` | string |  |  | Optional. The network security group to assign to the subnet |
+| `networkSecurityGroupNameResourceGroupName` | string | `[resourceGroup().name]` |  | Optional. Resource Group where NSGs are deployed, if different than VNET Resource Group. |
+| `privateEndpointNetworkPolicies` | string |  | `[Disabled, Enabled, ]` | Optional. enable or disable apply network policies on private end point in the subnet. |
+| `privateLinkServiceNetworkPolicies` | string |  | `[Disabled, Enabled, ]` | Optional. enable or disable apply network policies on private link service in the subnet. |
+| `routeTableName` | string |  |  | Optional. The route table to assign to the subnet |
+| `serviceEndpointPolicies` | array | `[]` |  | Optional. An array of service endpoint policies. |
+| `serviceEndpoints` | array | `[]` |  | Optional. The service endpoints to enable on the subnet |
+| `virtualNetworkName` | string |  |  | Required. The name of the parent virtual network |
+
+### Parameter Usage: `delegations`
+
+```json
+"delegations": [
+    {
+        "name": "sqlMiDel",
+        "properties": {
+            "serviceName": "Microsoft.Sql/managedInstances"
+        }
+    }
+]
+```
+
+### Parameter Usage: `serviceEndpoints`
+
+```json
+"serviceEndpoints": [
+    "Microsoft.EventHub",
+    "Microsoft.Sql",
+    "Microsoft.Storage",
+    "Microsoft.KeyVault"
+]
+```
+
+## Considerations
+
+The `privateEndpointNetworkPolicies` property must be set to disabled for subnets that contain private endpoints. It confirms that NSGs rules will not apply to private endpoints (currently not supported, [reference](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-overview#limitations)). Default Value when not specified is "Enabled".
+
+
+## Outputs
+
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `subnetAddressPrefix` | string | The address prefix for the subnet |
+| `subnetAddressPrefixes` | array | List of address prefixes for the subnet |
```

Comment:
@MrMCake , this one is still failing because it is optional and may not exist
```bash
The template output
     | 'subnetAddressPrefixes' is not valid: The language expression
     | property 'addressPrefixes' doesn't exist, available properties
     | are 'provisioningState, addressPrefix
```
We can just keep the first one as its a 'required' property and will always be available. The property may not exist hence its attempting to find it. Or maybe something using the `contains()` function and outputting `[]` if it doesn't exist?

Reply:
Added a condition

Reply:
And successfully tested